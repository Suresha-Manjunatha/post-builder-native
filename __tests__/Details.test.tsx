/** @format */

import { render } from "@testing-library/react-native";
import React from "react";
import Details from "../src/screens/details";

jest.mock("@react-navigation/native", () => {
	return {
		useNavigation: () => ({
			navigate: jest.fn(),
		}),
	};
});

describe("Details", () => {
	jest.useFakeTimers();

	const routeParamTest = {
		params: { item: "ok" },
	};

	it("renders without crashing", () => {
		render(<Details route={routeParamTest} />);
	});

	it("matches snapshot", () => {
		const tree = render(<Details route={routeParamTest} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("has 1 child", () => {
		const tree = render(<Details route={routeParamTest} />).toJSON();
		//@ts-ignore
		expect(tree?.children?.length).toBe(1);
	});

	it("has component with req testID", () => {
		const tree = render(<Details route={routeParamTest} />);
		//@ts-ignore
		expect(tree.getByTestId("details")).toBeTruthy();
	});
});
