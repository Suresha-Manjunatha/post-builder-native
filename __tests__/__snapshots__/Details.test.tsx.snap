// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details matches snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "padding": 10,
    }
  }
  testID="details"
>
  <Text>
     
    ok
  </Text>
</View>
`;
